CDCgov/prime-reportstream
# 6233 Fix submissions list returning multiple entries for one submission

This PR fixes the submissions list returning multiple entries for one submission.  I also got together with the UI team to verify all the fields they are using and updated the OpenAPI to match what is in the response.

Test Steps:
1. Check out this branch and run report stream.
2. Submit a report using your favorite sender.  Include the header `payloadname` with a custom report name, so you can see it in the response.
3. Do a GET on `api/waters/org/{{senderOrg}}/submissions` for the sender you used and verify 
   a. `reporItemCount` has the correct number of items
   b. `externalName` has your custom payloadname
   c. `sender` has the complete sender name as in <org>.<service>
   d. There is only one report per submission ID
4. Verify the OpenAPI spec matches this output.

Sample new API response:
```
[
    {
        "submissionId": 84,
        "timestamp": "2022-08-05T14:19:42.405Z",
        "externalName": "thepayloadname",
        "id": "b4deb5f4-569e-4a30-a1af-d1dac49ee240",
        "topic": "covid-19",
        "reportItemCount": 25,
        "httpStatus": 201,
        "sender": "careevolution.default"
    },
    {
        "submissionId": 23,
        "timestamp": "2022-08-04T18:34:43.729Z",
        "id": "c8e29164-3199-430f-850e-0b1b388c817a",
        "topic": "covid-19",
        "reportItemCount": 25,
        "httpStatus": 201,
        "sender": "careevolution.default"
    },
    {
        "submissionId": 21,
        "timestamp": "2022-08-04T18:34:35.003Z",
        "id": "aa7f1858-b57a-4cdf-9c34-bf946e8932a5",
        "topic": "covid-19",
        "reportItemCount": 25,
        "httpStatus": 201,
        "sender": "careevolution.default"
    },
```

## Changes
- fixed the submissions list returning multiple entries for one submission
- Fixed reportItemCount returning the correct number
- Fixed sender having the full sender name

## Checklist

### Testing

### Process

## Linked Issues
- #6233

## To Be Done

## Specific Security-related subjects a reviewer should pay specific attention to
